packages/celotool/src/cmds/local_testnet.ts

```diff
@@ -0,0 +1,381 @@
+      description: 'Local path to celo-blockchain repository.',
+      alias: ['localGeth', 'localgeth'],
+    })
+    .option('keep-data', {
+      type: 'boolean',
+      decription: 'Keep the data directory from any previous runs.',
+      alias: ['keepData', 'keepdata'],
+    })
+    .option('branch', {
+      type: 'string',
+      description: 'Branch name for remote celo-blockchain repository.',
+    })
+    .option('bootnode', {
+      type: 'boolean',
+      allowNo: true,
+      description: 'Create a bootnode and connect all nodes to it instead of to each other.',
+    })
+    .option('validators', {
+      type: 'number',
+      description: 'Number of validator nodes to create.',
+      default: 1,
+    })
+    .option('proxies', {
+      type: 'number',
+      description: 'Number of proxy nodes to create; assigned to the first n validators.',
+      default: 0,
+    })
+    .option('tx-nodes', {
+      type: 'number',
+      description: 'Number of transaction (i.e. non-validating full nodes) nodes to create.',
+      default: 0,
+      alias: ['txnodes', 'txNodes'],
+    })
+    .option('light-clients', {
+      type: 'number',
+      description: 'Number of light sync nodes to create.',
+      default: 0,
+      alias: ['lightClients', 'lightclients'],
+    })
+    .option('lightest-clients', {
+      type: 'number',
+      description: 'Number of lightest sync nodes to create.',
+      default: 0,
+      alias: ['lightestClients', 'lightestclients'],
+    })
+    .option('migrate', {
+      type: 'boolean',
+      description: 'Whether migrations should be run.',
+      default: true,
+      allowNo: true,
+    })
+    .option('migrate-to', {
+      type: 'number',
+      description: 'Maximum migration number to run. Defaults to running all migrations.',
+      alias: ['migrateTo', 'migrateto'],
+    })
+    .option('instances', {
+      type: 'string',
+      description: 'Manually enter a GethInstanceConfig[] json blob to add to the config.',
+      default: '[]',
+    })
+    .option('genesis-override', {
+      type: 'string',
+      description: 'Genesis configuration overrides as a GenesisConfig JSON blob.',
+      default: '{}',
+      alias: ['genesisOverride', 'genesisoverride'],
+    })
+    .option('migration-override', {
+      type: 'string',
+      description: 'Migration configuration overrides as a JSON blob.',
+      default: '{}',
+      alias: ['migrationOverride', 'migrationoverride'],
+    })
+}
+
+async function repl(config: GethRunConfig) {
+  const session = require('repl').start()
+  const formatName = (name: string) =>
+    name
+      .split('-')
+      .map((token, i) => (i === 0 ? token[0] : token[0].toUpperCase()) + token.slice(1))
+      .join('')
+
+  extend(session.context, {
+    Web3,
+    Admin,
+    testnet: config,
+    ...config.instances.reduce(
+      (o, instance) => ({
+        ...o,
+        [formatName(instance.name)]: {
+          web3: new Web3(getRpcUrl(instance)),
```

Comment:
this is amazing
wonder if it's worthwhile in future to add a contractkit here too

Reply:
I added a contractkit instance 👌